src/libraries/System.Private.CoreLib/src/System/IO/BufferedStream.cs

```diff
@@ -519,7 +539,7 @@ public override int Read(byte[] buffer, int offset, int count)
 
             // Ok. We can fill the buffer:
             EnsureBufferAllocated();
-            _readLen = _stream.Read(_buffer!, 0, _bufferSize);
+            _readLen = _stream.Read(_buffer!, 0, _buffer!.Length);
```

Comment:
Why change it here but not, for example, in the `if` check above?

Reply:
There are plenty of places where `_bufferSize` is used. Since I've not hit the shadow copy buffer code path in all of my benchmarks, I've decided to rever this change as it would require us to revisit all usages of `_bufferSize` and maybe the idea of shadow copy buffer itself.